patches/server/0009-Optimize-Network-Manager-and-add-advanced-packet-sup.patch

```diff
@@ -285,117 +278,100 @@ index 8f8b9aa731f2c97a3e32369bfd4d641a1f2ab412..f6e797f712f029a9bbe94739469c7464
          this.channel.pipeline().addBefore("prepender", "encrypt", new PacketEncrypter(MinecraftEncryption.a(1, secretkey)));
      }
  
 +    public boolean isConnected() { return this.g(); } // PandaSpigot - OBFHELPER
      public boolean g() {
          return this.channel != null && this.channel.isOpen();
      }
-@@ -314,9 +449,9 @@ public class NetworkManager extends SimpleChannelInboundHandler<Packet> {
+@@ -313,9 +447,9 @@ public class NetworkManager extends SimpleChannelInboundHandler<Packet> {
                  } else if (this.getPacketListener() != null) {
                      this.getPacketListener().a(new ChatComponentText("Disconnected"));
                  }
 -                this.i.clear(); // Free up packet queue.
-+                this.clearPacketQueue(); // PandaSpigot
++                clearPacketQueue(); // PandaSpigot
              } else {
 -                NetworkManager.g.warn("handleDisconnection() called twice");
 +                //NetworkManager.g.warn("handleDisconnection() called twice"); // PandaSpigot - Do not log useless message
              }
  
          }
-@@ -335,6 +470,8 @@ public class NetworkManager extends SimpleChannelInboundHandler<Packet> {
-             this.a = packet;
-             this.b = agenericfuturelistener;
-         }
-+        public Packet getPacket() { return this.a; } // PandaSpigot - OBFHELPER
-+        public GenericFutureListener<? extends Future<? super Void>>[] getGenericFutureListeners() { return this.b; } // PandaSpigot - OBFHELPER
-     }
+@@ -327,8 +461,8 @@ public class NetworkManager extends SimpleChannelInboundHandler<Packet> {
+ 
+     static class QueuedPacket {
  
-     // Spigot Start
+-        private final Packet a;
+-        private final GenericFutureListener<? extends Future<? super Void>>[] b;
++        private final Packet a; private final Packet<?> getPacket() { return this.a; } // PandaSpigot - OBFHELPER
++        private final GenericFutureListener<? extends Future<? super Void>>[] b; private final GenericFutureListener<? extends Future<? super Void>>[] getGenericFutureListeners() { return this.b; } // PandaSpigot - OBFHELPER
+ 
+         public QueuedPacket(Packet packet, GenericFutureListener<? extends Future<? super Void>>... agenericfuturelistener) {
+             this.a = packet;
 diff --git a/src/main/java/net/minecraft/server/Packet.java b/src/main/java/net/minecraft/server/Packet.java
-index fdc142b75b78289f01a9e9fed6f36432624735db..d60ec06b995eb477b555d222e99e208560656cef 100644
+index fdc142b75b78289f01a9e9fed6f36432624735db..8c9cd810779d1bbffb94dcab641602c05d19182d 100644
 --- a/src/main/java/net/minecraft/server/Packet.java
 +++ b/src/main/java/net/minecraft/server/Packet.java
-@@ -9,4 +9,37 @@ public interface Packet<T extends PacketListener> {
+@@ -9,4 +9,19 @@ public interface Packet<T extends PacketListener> {
      void b(PacketDataSerializer packetdataserializer) throws IOException;
  
      void a(T t0);
-+    
```

Comment:
Maybe revert those changes in the docs? The one from hpxfd looks a little better to me